### ISSUE TITLE
Crisp progurd rules for full mode

### ISSUE THREAD

**Issue by rezafaraji93** [crisp-im/crisp-sdk-android#96]
Would you please provide progurd rules for crisp?

**Comment by rezafaraji93** [comment]
I am using R8 full mode and ANR happens because of Crisp

**Comment by Doc1faux** [comment]
Hi @rezafaraji93 and thank your for your feedback.

I didn't tested yet minification with [R8 full mode](https://r8.googlesource.com/r8/+/refs/heads/master/compatibility-faq.md#r8-full-mode) as it was an experimental feature.
However, I've just tested it and indeed it crashes... 
But this isn't due directly to our SDK but from [Gson](https://github.com/google/gson) and [Retrofit](https://github.com/square/retrofit) libraries.
They miss to export proguard rules:
- [R8 full mode - Gson rules](https://r8.googlesource.com/r8/+/refs/heads/master/compatibility-faq.md#gson-with-full-mode)
- [R8 full mode - Retrofit rules](https://r8.googlesource.com/r8/+/refs/heads/master/compatibility-faq.md#retrofit)

For Retrofit, from the `2.6.4` version we're using (to keep 4.4 compatibility) to the latest `2.9.0` release, they only export the following rules:
```
-keepattributes Signature
-if interface * { @retrofit2.http.* <methods>; }
-keep,allowobfuscation interface <1>
```
But more are needed according to their [master branch](https://github.com/square/retrofit/blob/master/retrofit/src/main/resources/META-INF/proguard/retrofit2.pro), so the following ones should also be exported:
```
# Keep generic signature of Call, Response (R8 full mode strips signatures from non-kept items).
-keep,allowobfuscation,allowshrinking interface retrofit2.Call
-keep,allowobfuscation,allowshrinking class retrofit2.Response
```
And if Kotlin is used, this one should be exported also:
```
# With R8 full mode generic signatures are stripped for classes that are not
# kept. Suspend functions are wrapped in continuations where the type argument
# is used.
-keep,allowobfuscation,allowshrinking class kotlin.coroutines.Continuation
```

So we end up exporting these proguard rules:
```
# Gson
-keep class com.google.gson.reflect.TypeToken { *; }
-keep class * extends com.google.gson.reflect.TypeToken
# Retrofit
-keep,allowobfuscation,allowshrinking interface retrofit2.Call
-keep,allowobfuscation,allowshrinking class retrofit2.Response
# Retrofit, only if Kotlin is used
-keep,allowobfuscation,allowshrinking class kotlin.coroutines.Continuation
```

I commited a fix for this but whilw waiting a new release, you already can apply these rules for your project, it should work ;)

**Comment by Doc1faux** [comment]
Fixed in `1.0.10` and released :)
